### PR TITLE
Run tests on docker if docker image exists

### PR DESCRIPTION
**Pull request by kaituo**
*Issue #, if available:*

*Description of changes:*

Previously, CI ran into issues when the docker image was not out yet after a version upgrade. This PR fixes it by using an environment variable to track whether the docker image exists or not.

This PR also removes test-workflow.yml since CI.yml can cover its functionality.

Testing done:
1. Verified in github docker tests skip if no docker image exists.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -48,15 +52,20 @@ jobs:
             echo "RUN /usr/share/elasticsearch/bin/elasticsearch-plugin install --batch file:/tmp/opendistro-anomaly-detection-$plugin_version.zip" >> Dockerfile
 
             docker build -t odfe-ad:test .
+            echo "::set-env name=imagePresent::true"
+          else
+            echo "::set-env name=imagePresent::false"
```
Comment (yizheliu-amazon) [inline .github/workflows/CI.yml:57]: if it is false, will the CI workflow fail?

Comment (kaituo) [reply .github/workflows/CI.yml:57]: no, it won't.  See the workflow ran as part of the PR.

Comment (yizheliu-amazon) [reply .github/workflows/CI.yml:57]: hmmm, should we fail workflow if docker image doesn't exist? It is a dependency to the workflow. If any PR comes when docker image is not there, we may miss the unit test as protection here.

Comment (kaituo) [reply .github/workflows/CI.yml:57]: unit tests still runs.  Only docker related tests won't run.

Comment (yizheliu-amazon) [reply .github/workflows/CI.yml:57]: I see. IT won't run if docker is missing. Will it be a concern if PR which may break IT gets merged, but not caught in CI given docker is missing?

Comment (kaituo) [reply .github/workflows/CI.yml:57]: IT runs as well. Just won't run in a docker environment.

Comment (yizheliu-amazon) [reply .github/workflows/CI.yml:57]: okay.

**Pull request merged**